sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1110,6 +1110,17 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     }
     assert(e.getMessage.contains("Cannot create encoder for Option of Product type"))
   }
+
+
+  test ("SPARK-17460: the sizeInBytes in Statistics shouldn't overflow to a negative number") {
+    // If the sizeInBytes in Statistics exceeds the limit of an Int,
+    // it should convert to a BigInt instead of overflowing to a negative number
```

Comment:
How about? 
`Since the sizeInBytes in Statistics could exceed the limit of an Int, we should use BigInt instead of Int for avoiding possible overflow`